src/buffers/RecordingBuffer.cpp

```diff
@@ -79,11 +84,15 @@ int RecordingBuffer::Read(byte *buffer, size_t length)
   if (dataRead == 0 && m_isLive)
   {
     kodi::Log(ADDON_LOG_DEBUG, "%s:%d: %lld %lld", __FUNCTION__, __LINE__, m_inputHandle.GetLength() , m_inputHandle.GetPosition());
-    int64_t position = m_inputHandle.GetPosition();
-    Buffer::Close();
-    Buffer::Open(m_recordingURL, ADDON_READ_NO_CACHE);
-    Seek(position, 0);
-    dataRead = (int) m_inputHandle.Read(buffer, length);
+    const int64_t position = m_inputHandle.GetPosition();
+    const time_t startTime = time(nullptr);
+    do {
+      Buffer::Close();
+      std::this_thread::sleep_for(std::chrono::milliseconds(2000));
+      Buffer::Open(m_recordingURL);
+      Seek(position, 0);
+      dataRead = static_cast<int>(m_inputHandle.Read(buffer, length));
+    } while (dataRead == 0 && time(nullptr) - startTime < 5);
```

Comment:
Do you want to calculate time(nullptr) on each loop or would once before the loop be sufficient?

Reply:
On each loop, I don't want to wait more than 5 seconds, and in case at real EOF the read doesn't return -1 I want to exit.

Reply:
Ah, sorry, makes sense. NVM